src/gazebo_mavlink_interface.cpp

```diff
@@ -767,20 +743,120 @@ void GazeboMavlinkInterface::IRLockCallback(IRLockPtr& irlock_message) {
 }
 
 void GazeboMavlinkInterface::VisionCallback(OdomPtr& odom_message) {
-  mavlink_vision_position_estimate_t sensor_msg;
-  sensor_msg.usec = odom_message->usec();
-  // convert from ENU to NED
-  sensor_msg.x = odom_message->y();
-  sensor_msg.y = -odom_message->x();
```

Comment:
This is what is causing me confusion: from the local ENU to local NED, you swap X with Y and negate Z. For the above to happen, it means you have the position on the FLU body frame, which means you first have to transform to body FRD (180º roll rotation) and then apply the transform from ENU to NED.

Reply:
Still, something is not right here, cause if we apply the 180º rotation on Roll, it negates Y and Z. So this transform is not making sense to me, even though it is working.